Framework/Core/src/DataSampling.cxx

```diff
@@ -59,14 +59,7 @@ void DataSampling::GenerateInfrastructure(WorkflowSpec& workflow, const std::str
     bool dataFound = false;
     for (const auto& dataProcessor : workflow) {
       for (const auto& externalOutput : dataProcessor.outputs) {
-        InputSpec candidateInputSpec{
-          "doesnt-matter", //externalOutput.binding.value,
-          externalOutput.origin,
-          externalOutput.description,
-          externalOutput.subSpec,
-          externalOutput.lifetime
-        };
-
+        InputSpec candidateInputSpec = DataSpecUtils::matchingInput(externalOutput);
```

Comment:
This breaks workflow generation when `externalOutput` has no binding - I get: In spec Dispatcher input specification 0 requires binding, description and origin to be fully specified. Could you put there something like `candidateInputSpec.binding = "doesnt-matter";` or make binding nonobligatory? Dispatcher doesn't use bindings anyway, it iterates on InputRecord.

Reply:
You mean like this? Or you think matchingInput is to be changed?

Reply:
yes, like this, thank you